ynput/OpenPype
# Maya: Reduce amount of code for Collect Looks

## Changelog Description

- Refactor `get_file_node_files` because popping from `paths` by index should have been done in reversed order anyway. It's now changed to not need popping at all.
- Removed unused `RENDERER_NODE_TYPES` and if-branch which collected `node_attrs` list which was unused + collected members which was also done outside of the if branch and thus generated no extra data.
- Collected all materials from look set attributes at once instead of multiple queries
- Collected all file nodes in history from a single query instead of per type
- Restructured assignment of `instance.data["resources"]` to be more readable
- Cached `PXR_NODES` only ones (Note: plugin load is checked on discovery of the collect look plugin) instead of querying plugin load and its nodes per file node per attribute
- Removed some debug logs or combined some messages

## Additional info

Also mentioned on Discord [here](https://discord.com/channels/517362899170230292/517382145552154634/1126252092202430555)

## Testing notes:

Preferably publish looks with different file node types (e.g. `file` and `aiImage` with and without UDIMs or sequences) for each renderer type:

1. Publish looks
    - [x] arnold (mtoa)
    - [x] redshift
    - [x] renderman
    - [x] v-ray

2. Ensure look assignments still work as intended 
    - [x] shader assignments
    - [x] render stat attributes on meshes 
    - [x] (Redshift) mesh parameters object sets, etc.
